src/interfaces/chain.h

```diff
@@ -83,6 +83,7 @@ struct BlockInfo {
     const uint256& hash;
     const uint256* prev_hash = nullptr;
     int height = -1;
+    int64_t time = 0;
```

Comment:
In 1a9a349a026af2865e5446091f1e072b7a845819 "wallet: skip block scan if block was created before wallet birthday"

`CBlock` already has a `GetBlockTime()`, so I don't think it's necessary to include the time as a separate field here.

However I'm not sure that we should be using the block's timestamp for this time rather than MTP. With the block timestamp, we could ignore a more recent block that has a timestamp older than its parent which we wouldn't ignore. That would be kinda bad.

Reply:
> CBlock already has a GetBlockTime(), so I don't think it's necessary to include the time as a separate field here.

Yeah, that works here. The new field was because I initially implemented this on the filters-only sync branch. Where I have no blocks, only headers and filters.

> However I'm not sure that we should be using the block's timestamp for this time rather than MTP. With the block timestamp, we could ignore a more recent block that has a timestamp older than its parent which we wouldn't ignore. That would be kinda bad.

Hmm, wasn't the timestamp window created to extend the grace period enough so we can use the block time directly?.
In other words, MTP moves the block time an hour back, and we have a grace period of two hours. If the two hours period concerns you, to not be forced to calculate the MTP for every block, what about doubling the grace period for IBD scanning?.
Normally, it would just mean checking ~12 more blocks.

Reply:
> Hmm, wasn't the timestamp window created to extend the grace period enough so we can use the block time directly?.

AFAIK it exists for accepting blocks with future timestamps, and we only use it in the wallet because it's convenient.

But my comment isn't really about the accuracy of the timestamp, but rather about the possibility of skipping a block after we have already started scanning. A block's timestamp can be before its parent. The code as written would let us skip scanning a block even if we had scanned its parent. The reason to use MTP is because it can only move forward. The MTP of a block can never be less than the MTP of its parent, it can be equal, but that's fine for us. In `RescanFromTime`, we find blocks using a different metric, `GetBlockTimeMax()`. This is calculated as the maximum of the current block's timestamp and it's parent's timestamp, so this too also can only move forward. If MTP is too slow to calculate, we could use `GetBlockTimeMax()` instead as it has that same property.

Reply:
> But my comment isn't really about the accuracy of the timestamp, but rather about the possibility of skipping a block after we have already started scanning. A block's timestamp can be before its parent. The code as written would let us skip scanning a block even if we had scanned its parent. The reason to use MTP is because it can only move forward. The MTP of a block can never be less than the MTP of its parent, it can be equal, but that's fine for us.

Yeah, where I was going is that we really don't care about those possible skipped blocks if the grace period is large enough. We know that the wallet will not find anything there, it's just to ensure that all blocks created 2 hours before the wallet birth time will be scanned.
In other words,  The idea was to extend the window enough to not worry about unordered timestamps. E.g. going to the extreme, even if the grace period is one or two days, is much better than scanning the entire chain as we do now.

But.. point taken. The `GetBlockTimeMax` idea sounds nicer. It doesn't require any extra calculation like MTP, nor grace period extension (which makes the wallet scan extra blocks). Just add another member into the `BlockInfo` struct and done.